### PR TITLE
fix(queries): standardise on "results" over "result"

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Some query nodes return their results as `result`, some as `results`.

## Changes

Standardises all on `results`, as this is what the currently publicly exposed `EventsQuery` and `HogQLQuery` use. 

Also, we almost always have more than one "result", so plural makes sense to me.

## How did you test this code?

The HogQL powered lifecycle insight loaded. Both in its cached and refreshed form... though I'm not sure what I don't know about all of this, so 👀 appreciated!

### CODE REVIEW

**Review by thmsobrmlr — COMMENTED** [review]
This makes sense and the change in `insightDataLogic` should be enough to ensure compatibility, though I'll check in detail later.

**Comment by mariusandra** [comment]
Thanks! I'm likely going to merge this into my [PersonsQuery PR](https://github.com/PostHog/posthog/pull/17764), as the pydantic validation errors about result vs results in the QueryResponse was what triggered this PR.

File: frontend/src/scenes/insights/insightDataLogic.ts [hunk 1]
```diff
@@ -147,6 +147,14 @@ export const insightDataLogic = kea<insightDataLogicType>([
                 }
             },
         ],
+
+        insightData: [
+            (s) => [s.insightDataRaw],
+            (insightDataRaw): Record<string, any> => {
+                // :TRICKY: The queries return reuslts as `results`, but insights expect `result`
```
Comment (thmsobrmlr) [inline frontend/src/scenes/insights/insightDataLogic.ts:154]:
```suggestion
                // :TRICKY: The queries return results as `results`, but insights expect `result`
```

**Review by thmsobrmlr — APPROVED** [review]
Found one place where we need to adapt the logic https://github.com/PostHog/posthog/blob/master/frontend/src/queries/nodes/DataNode/dataNodeLogic.ts#L69. Caching and all other places I though of look fine 🤞

**Pull request merged**